wordpress-mobile/wellsql
# [Build] Upgrade Gradle to 8.2.1 and AGP to 8.0.2

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.0.2](https://developer.android.com/build/releases/gradle-plugin#agp-8-0-2).

FYI: The [8.1.1](https://docs.gradle.org/8.1.1/release-notes.html) Gradle upgrades had to happen in 2 parts (96baa77d79c6a64b44801f9336984ee911ffaa16 and 1260b8d86f4ae6879451cfa2a1f73cc6cd8196f7). This is because of the breaking changes that this upgrade introduces, including the `AGP` upgrade and its associated breaking changes (namespace, build option, etc).

PS: This Gradle and AGP upgrade was done in order to align this repo with the rest of such [WordPress-Mobile](https://github.com/wordpress-mobile) Android library related repos. However, mind the fact that this work here is not necessary to complete the aforementioned project, just some extra work to help with the wider alignment efforts.

-----

🚫 Blocked By:


<details>
    <summary>CI Failure</summary>

```
[2023-07-17T13:57:38Z] FAILURE: Build failed with an exception.
--
  | [2023-07-17T13:57:38Z]
  | [2023-07-17T13:57:38Z] * Where:
  | [2023-07-17T13:57:38Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-0c1f8588a78c5ea3a-1/automattic/wellsql/well-sample/build.gradle' line: 2
  | [2023-07-17T13:57:38Z]
  | [2023-07-17T13:57:38Z] * What went wrong:
  | [2023-07-17T13:57:38Z] An exception occurred applying plugin request [id: 'com.android.application', version: '8.0.2']
  | [2023-07-17T13:57:38Z] > Failed to apply plugin 'com.android.internal.application'.
  | [2023-07-17T13:57:38Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-17T13:57:38Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-17T13:57:38Z]       You can try some of the following options:
  | [2023-07-17T13:57:38Z]        - changing the IDE settings.
  | [2023-07-17T13:57:38Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-17T13:57:38Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Move package from android manifest to build files](https://github.com/wordpress-mobile/wellsql/commit/1eb6cb489f037c6f8915139aefe8c498d6181815)
2. [Use new publishing dsl to configure publication variants](https://github.com/wordpress-mobile/wellsql/commit/63c716a6dac263411729d0d95c22c753581769a4)

AGP Upgrade Assistant List:

1. [Remove ignored configuration properties](https://github.com/wordpress-mobile/wellsql/pull/25/commits/baf8f5dfc6ceb9043981924e33774d0ff363c1de)
2. [Migrate dsl properties from lint options to lint](https://github.com/wordpress-mobile/wellsql/pull/25/commits/b60a14563ca417cdf2cbf4053472c9851efa0968)

-----

## Associated clients (`N/A`)

FYI: `wellsql` is not part of a composite build, neither for `WPAndroid`, `WCAndroid` clients or the `FluxC` library.

-----

## To test:

- There is nothing much to test here (maybe that the `publishing` is still working as expected).
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could run `./gradlew build` locally and verify that  it is successful (see [previous](https://github.com/wordpress-mobile/wellsql/pull/24) such update).

-----

## Merge instructions

- [ ] In coordination with all the other library and client PRs, merge this PR only when every PR is ready to be merged. FYI: This is a nice to have step here, but not a necessary step per se since `wellsql` is not part of a composite build.
